Consistent hard crash on windows when started from nushell

### Summary

When editing the attached file in a specific way this gives a crash without any errors. Only seems to happen on windows when Helix is started from nushell.

[crash.md](https://github.com/helix-editor/helix/files/13363272/crash.md)


### Reproduction Steps

<!-- Ideally provide a key sequence and/or asciinema.org recording. --> 

I tried this:

1. Open a terminal and start nu shell: `nu`
2. `$env.RUST_BACKTRACE = 1` (per instructions for detailed logging)
3. `hx -vv -c /dev/null crash.md` (open attached crashing file in hx with default config)
4. `:100` (goto line 100)
5. `o` (open a new line)
6. start typing a moderate length sentence

This happened
![image](https://github.com/helix-editor/helix/assets/1030961/25d8ef90-ab4f-47a9-a9da-5451e9d51519)

Let me know if I can help trouble shooting. I'm compiling from source, so I can easily add compilation flags or something. But I'm no Rust programmer nor have I a debugger installed/setup.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-11-15T11:16:16.288 helix_view::editor [ERROR] Failed to initialize the language servers for `source.md` - `marksman` { cannot find binary path }
2023-11-15T11:16:16.296 helix_view::editor [DEBUG] editor status: Loaded 1 file.
2023-11-15T11:16:16.296 helix_tui::backend::crossterm [DEBUG] The keyboard enhancement protocol is not supported in this terminal (checked in 0ns)
2023-11-15T11:16:16.298 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:16.302 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:16.527 helix_term::application [DEBUG] received editor event: IdleTimer
2023-11-15T11:16:17.705 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:17.913 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:17.960 helix_term::application [DEBUG] received editor event: IdleTimer
2023-11-15T11:16:17.997 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:18.141 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:18.326 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:20.702 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:20.940 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:21.095 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:21.158 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:21.189 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:21.200 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:21.231 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:21.236 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:21.288 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-11-15T11:16:21.300 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0```
```
</details>


### Platform

Windows 11

### Terminal Emulator

wezterm, cmd.exe

### Installation Method

compiled from source

### Helix Version

a98ad137

--- a/languages.toml
+++ b/languages.toml
@@ -1481,7 +1481,7 @@ block-comment-tokens = { start = "<!--", end = "-->" }
 
 [[grammar]]
 name = "markdown"
-source = { git = "https://github.com/MDeiml/tree-sitter-markdown", rev = "aaf76797aa8ecd9a5e78e0ec3681941de6c945ee", subpath = "tree-sitter-markdown" }
+source = { git = "https://github.com/tree-sitter-grammars/tree-sitter-markdown", rev = "62516e8c78380e3b51d5b55727995d2c511436d8", subpath = "tree-sitter-markdown" }
 
 [[language]]
 name = "markdown.inline"
@@ -1492,7 +1492,7 @@ grammar = "markdown_inline"
 
 [[grammar]]
 name = "markdown_inline"
-source = { git = "https://github.com/MDeiml/tree-sitter-markdown", rev = "aaf76797aa8ecd9a5e78e0ec3681941de6c945ee", subpath = "tree-sitter-markdown-inline" }
+source = { git = "https://github.com/tree-sitter-grammars/tree-sitter-markdown", rev = "62516e8c78380e3b51d5b55727995d2c511436d8", subpath = "tree-sitter-markdown-inline" }
 
 [[language]]
 name = "dart"

